AK/StreamBuffer.h

```diff
@@ -124,8 +125,113 @@ class StreamBuffer {
     u8* m_data { nullptr };
 };
 
+class StreamSeekbackBuffer {
+public:
+    StreamSeekbackBuffer(size_t max_seekback_distance, size_t max_back_reference_length, double optimization_factor = 2.0)
```

Comment:
It just makes the backing array larger. Having larger storage means that we are less likely to wrap around (which is complicated => slow).

Reply:
Looks to be the same reasoning as with `max_back_reference_length`, so let's wrap the discussion into the other comment thread.

Reply:
Nah, `optimization_factor` is a bit different from `max_back_reference_length`. The algorithm just won't give correct results if we provide backing storage of less than `max_back_reference_length + max_seekback_distance` bytes (because I indirectly assume copying doesn't overwrite existing data). On the other hand, any `optimization_factor` no less than 1 will work.